### PR TITLE
:sparkles: Add noopdelete option to extension api

### PR DESCRIPTION
**Pull request by gallettilance**
# Description

Solves #723 

## Reviewer Checklist

- [ ] API Go Documentation
- [ ] Tests: Unit Tests (and E2E Tests, if appropriate)
- [ ] Comprehensive Commit Messages
- [ ] Links to related GitHub Issue(s)

### CODE REVIEW

File: internal/controllers/extension_controller.go [hunk 1]
```diff
@@ -432,6 +432,10 @@ func (r *ExtensionReconciler) GenerateExpectedApp(o ocv1alpha1.Extension, bundle
 		},
 	}
 
+	if o.Spec.NoopDelete {
```
Comment (joelanford) [inline internal/controllers/extension_controller.go:435]: No need for a conditional here, right?

**Comment by grokspawn** [comment]
I ✨ the PR title to pass the pr-title workflow, based on PR contents.  This repo enforces naming along these lines: 
    * ⚠ (:warning:, major/breaking change)
    * ✨ (:sparkles:, minor/compatible change)
    * 🐛 (:bug:, patch/bug fix)
    * 📖 (:book:, docs)
    * 🌱 (:seedling:, other)
    
   If that's not correct, please choose the option that best fits.

**Comment by tmshort** [comment]
`make verify` needs to be run before commiting.

**Comment by joelanford** [comment]
I'm curious if there's a reason we're adding a new feature to the extension API, given that we're planning to focus on the cluster extension API and temporarily remove this API.

Should we just make this change directly on the ClusterExtension instead?

File: api/v1alpha1/extension_types.go [hunk 2]
```diff
@@ -74,6 +74,9 @@ type ExtensionSpec struct {
 	// paused controls the management state of the extension. If the extension is paused, it will be ignored by the extension controller.
 	Paused bool `json:"paused,omitempty"`
 
+	// NoopDelete controls what is done with the associated resources. If set to true they will not be deleted.
+	NoopDelete bool `json:"noopdelete,omitempty"`
```
Comment (varshaprasad96) [inline api/v1alpha1/extension_types.go:78]: Nit: We could add validation to this with KB markers, setting it to optional, so we don't worry about the default value, then directly assign it to App's spec.

**Review by varshaprasad96 — COMMENTED** [review]
Looks good! /lgtm from my end, after addressing Joe's review. 

Regarding surfacing this on the status - looking deeper, it doesn't seem necessary imo. We don't specify anything explicitly for paused state either. Also this info may be useful only when the user tries to delete the relevant extension, at which point we are not going to persist the object itself. Just cross-checked, looks like [PackageInstall](https://github.com/carvel-dev/kapp-controller/blob/a8f75d9c13baa2f63c43a1aa6da845458091629f/pkg/apis/packaging/v1alpha1/package_install.go#L103) doesn't have it too.

File: api/v1alpha1/extension_types.go (hunk 2 above)
Comment (gallettilance) [reply api/v1alpha1/extension_types.go:81]: done :)

**Comment by joelanford** [comment]
Close or update/rebase for `ClusterExtension`?

**Pull request closed without merging**